### PR TITLE
CMS 2020 queries

### PR DESCRIPTION
**Pull request by rviscomi**
Queries for the CMS 2020 chapter (#915)

### CMS adoption
- [x] % of desktop and mobile websites that use a CMS
  - compare with 2019

### Top CMSs
- [x] top ~10 CMSs for desktop and mobile
  - compare changes in individual CMS adoption with 2019

### CMS composition
- [x] distribution of total bytes per CMS page
  - [x] environmental impact in CO2 (using [carbonapi](https://gitlab.com/wholegrain/carbon-api-2-0/-/blob/master/includes/carbonapi.php#L342))
- [x] distribution of  total requests CMS page
- [x] distribution of bytes per resource type on CMS pages
- [x] distribution of 3P bytes per CMS page
- [x] distribution of 3P requests per CMS page
- [x] distribution of image bytes per CMS page
- [x] adoption of image formats on CMS pages

#### Core Web Vitals compliance by CMS
- [x] % of pages that pass the LCP assessment grouped by CMS
- [x] % of pages that pass the FID assessment grouped by CMS
- [x] % of pages that pass the CLS assessment grouped by CMS
- [x] % of pages that pass the LCP, FID, and CLS assessments grouped by CMS

#### Average distribution performance of Core Web Vitals by CMS

- [x] average good/NI/poor distribution of LCP per CMS
- [x] average good/NI/poor distribution of FID per CMS
- [x] average good/NI/poor distribution of CLS per CMS

#### Lighthouse
- [x] median Lighthouse Performance category score per CMS
- [x] median Lighthouse PWA category score per CMS
- [x] median Lighthouse Accessibility category score per CMS

cc @GregBrimble @alexdenning @sirjonathan @ernee

### CODE REVIEW

**Comment by max-ostapenko** [comment]
For WebVitals and Lighthouse perf you can re-use queries from Performance chapter #1091.
You'll just need to apply origins filter.

**Comment by GregBrimble** [comment]
Looks like we don't have `httparchive.technologies.2020_07_01_mobile` yet.

I've made a start with these queries and will finish the rest off this evening.

**Comment by GregBrimble** [comment]
[As discussed in the doc](https://docs.google.com/document/d/1XL87z5MOk7QDLhpFsaMWT3DaIXdOQV32qs57noGwDVQ/edit?disco=AAAAJ1Bq3uE), figuring out whether a website is considered green or not by the Green Web Foundation requires a bit more work. For now, assuming everything is not green, I can pull data like this:

```
[
  {
    "percentile": "10",
    "client": "desktop",
    "requests": "39",
    "mbytes": "0.72",
    "co2grams": "0.4571399928809842"
  },
  {
    "percentile": "10",
    "client": "mobile",
    "requests": "38",
    "mbytes": "0.68",
    "co2grams": "0.43123380775738046"
  },
  {
    "percentile": "25",
    "client": "desktop",
    "requests": "58",
    "mbytes": "1.27",
    "co2grams": "0.803129719136632"
  },
  {
    "percentile": "25",
    "client": "mobile",
    "requests": "56",
    "mbytes": "1.23",
    "co2grams": "0.7774688988400158"
  },
  {
    "percentile": "50",
    "client": "desktop",
    "requests": "87",
    "mbytes": "2.3",
    "co2grams": "1.4555945829872508"
  },
  {
    "percentile": "50",
    "client": "mobile",
    "requests": "85",
    "mbytes": "2.26",
    "co2grams": "1.4270840340980793"
  },
  {
    "percentile": "75",
    "client": "desktop",
    "requests": "129",
    "mbytes": "4.17",
    "co2grams": "2.634180113511393"
  },
  {
    "percentile": "75",
    "client": "mobile",
    "requests": "127",
    "mbytes": "4.06",
    "co2grams": "2.56586680793378"
  },
  {
    "percentile": "90",
    "client": "desktop",
    "requests": "185",
    "mbytes": "7.31",
    "co2grams": "4.61789385064505"
  },
  {
    "percentile": "90",
    "client": "mobile",
    "requests": "181",
    "mbytes": "6.95",
    "co2grams": "4.390533496772987"
  }
]
```

Is this the type of data you're after @alexdenning ?

**Comment by GregBrimble** [comment]
That's all but the Web Vitals and Lighthouse queries done. Still need to dig into the [Performance chapter's PR](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1091) and figure out how to grab what we need from there.

**Comment by rviscomi** [comment]
@GregBrimble I pushed a commit to your queries to update them to the August dataset and made some minor formatting changes. The `almanac.summary_requests` dataset only contains 2019 data so I updated it to use `almanac.requests` instead. Overall your queries look great, nice work!

**Comment by rviscomi** [comment]
Also @alexdenning @GregBrimble FYI: https://github.com/HTTPArchive/almanac.httparchive.org/pull/1091#issuecomment-684980916

The Lighthouse Performance audits/scores are unreliable in the August dataset so any metrics that depend on them should be deferred until the corrected data in the September dataset are available (another month 😢).

File: sql/2020/15_CMS/core_web_vitals_distribution.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  SUM(large_cls) / (SUM(small_cls) + SUM(medium_cls) + SUM(large_cls)) AS poor_cls,
+FROM
+  `chrome-ux-report.materialized.device_summary`
+JOIN (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    app AS cms
+  FROM
+    `httparchive.technologies.2019_08_01_*`
```
Comment (denar90) [inline sql/2020/15_CMS/core_web_vitals_distribution.sql:30]:
```suggestion
    `httparchive.technologies.2020_08_01_*`
```

File: sql/2020/15_CMS/core_web_vitals_passing.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      IS_NON_ZERO(small_cls, medium_cls, large_cls), origin, NULL))) AS pct_good_cwv
+FROM
+  `chrome-ux-report.materialized.device_summary`
+JOIN (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    app AS cms
+  FROM
+    `httparchive.technologies.2019_08_01_*`
```
Comment (denar90) [inline sql/2020/15_CMS/core_web_vitals_passing.sql:49]:
```suggestion
    `httparchive.technologies.2020_08_01_*`
```

**Review by Tiggerito — APPROVED** [review]
Looks all nice to me (with my limited knowledge)

**Review by DoubleRR7244 — APPROVED** [review]

**Comment by GregBrimble** [comment]
Thanks @rviscomi for finishing off for me there. I've requested edit access to [the Results sheet](https://docs.google.com/spreadsheets/d/1vTf459CcCbBuYeGvgo-RSidppR62SfM-VTkW-dfS3K4/edit#gid=908727245), and once I get access, I'll run the queries that we haven't yet added.

Thanks also to @denar90 (and other reviewers), I've committed your suggested changes.

**Pull request merged**